scripts/modules_auto_pr/modules_auto_pr.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+####################################################################################################
+#                                            VARIABLES                                             #
+####################################################################################################
+
+# Variables (replace these with your actual values)
+DATE=$(date +"%Y-%m-%d_%Hh%M")
+REPO_URL="git@github.com:lbarraga/token-test.git"
+BASE_BRANCH="main"
+BRANCH_NAME="new-feature-branch_$DATE"
+REPO_NAME="token-test_$DATE"
+REPO_PATH="/tmp/$REPO_NAME"
+FILE_NAME="newfile.txt"
+COMMIT_MESSAGE="Add newfile.txt"
+PR_TITLE="Add new file as part of feature"
+PR_BODY="This PR adds a new file to the repository."
+
+# Use GitHub CLI for authentication
+GH_TOKEN="SAMPLE_TOKEN" # Test token, will not be used in production
+
+echo_info()    { echo -e "\e[32m$0: [INFO] $1\e[0m"; }
+echo_warning() { echo -e "\e[33m$0: [WARNING] $1\e[0m"; }
+echo_error()   { echo -e "\e[31m$0: [ERROR] $1\e[0m"; }
+echo_error_exit() {
+  echo_error "$1"
+  rm -rf "$REPO_PATH"
+  exit 1
+}
+
+####################################################################################################
+#                                             SCRIPT                                               #
+####################################################################################################
+
+echo_info "Logging in to GitHub..."
+gh auth login --with-token <<< "$GH_TOKEN" || echo_error_exit "Failed to log in to GitHub"
+
+# Check if the repo directory already exists and delete it if so
+if [ -d "$REPO_PATH" ]; then
+    echo_error "Directory $REPO_PATH already exists."
+    echo_error "Please remove it manually:"
+    echo_error "  $ rm -rf $REPO_PATH"
+    exit 1
+fi
+
+echo_info "Cloning repo $REPO_URL..."
+git clone $REPO_URL "$REPO_PATH"  || echo_error_exit "Failed to clone repo $REPO_URL"
+
+echo_info "Navigating to repo directory..."
+cd "$REPO_PATH"                   || echo_error_exit "Failed to cd into $REPO_PATH"
+
+echo_info "Checking out base branch $BASE_BRANCH..."
+git checkout -b "$BRANCH_NAME"    || echo_error_exit "Failed to create branch $BRANCH_NAME"
+
+# run available_software.py
+echo_info "Running available_software.py..."
+cd scripts/available_software/    || echo_error_exit "Failed to cd into scripts/available_software/"
+python -m venv venv               || echo_error_exit "Failed to create virtual environment"
+source venv/bin/activate          || echo_error_exit "Failed to activate virtual environment"
+pip install requests              || echo_error_exit "Failed to install requests"
```

Comment:
so `gh` is present, but `python3-requests` is not?

Reply:
Took me some time to figure out why the requirements where not installing properly. It seems I placed too much faith into copilot tab completion

Reply:
This will most likely run on the vsc account of @boegel. He told me I could assume the `gh` command is present.

Reply:
Yeah, if `gh` is not available, just let the script fail early on.